tests/models/apifilter.spec.ts

```diff
@@ -4,11 +4,35 @@ import { ApiFilter } from '../../src/models/apifilter';
 
 describe('ApiFilter', () => {
 
-  it('check ApiFilter basic functionality', () => {
+  it('check ApiFilter add filter', () => {
     let filter: ApiFilter = new ApiFilter();
     filter.addFilter('foo', 'bar');
     filter.addFilter('abc', '5');
     expect(filter.filterString()).toEqual('?foo=bar&abc=5');
   });
 
+  it('check ApiFilter add same filter', () => {
+    let filter: ApiFilter = new ApiFilter();
+    filter.addFilter('foo', 'bar');
+    filter.addFilter('abc', '5');
+    filter.addFilter('abc', '9');
+    expect(filter.filterString()).toEqual('?foo=bar&abc=5&abc=9');
+  });
+
+  it('check ApiFilter remove filter', () => {
+    let filter: ApiFilter = new ApiFilter();
+    filter.addFilter('foo', 'bar');
+    filter.addFilter('abc', '5');
+    filter.removeFilter('foo');
+    expect(filter.filterString()).toEqual('?abc=5');
+  });
+
+  it('check ApiFilter add unique filter', () => {
+    let filter: ApiFilter = new ApiFilter();
+    filter.addFilter('foo', 'bar');
+    filter.addFilter('abc', '5');
+    filter.addFilter('abc', '9', true);
```

Comment:
You should test what happens if I use remove=true but the value does not exist. Confirm it is ok